src/Validation/CaValidation.php

```diff
@@ -59,3 +108,101 @@ public static function personId($check)
         throw new NotImplementedException(__d('localized', '%s Not implemented yet.'));
     }
 }
+
+class InvalidNumberRule
```

Comment:
Not that a separate class over regexp wouldn't be a viable option, but this looks like a snowflake between all the other validation rules if not applied on a more discussed and global level.